src/lib/debug.jsx

```diff
@@ -0,0 +1,138 @@
+  flag('debug.notes.service', true)
+  flag('debug.notes.collabProvider', true)
+  flag('debug.notes.channel', true)
+  flag('debug.notes.noteId', true)
+  flag('debug.notes.doc', true)
+  flag('debug.notes.file', true)
+  flag('debug.notes.returnUrl', true)
+  flag('debug.notes.dirtyRef', true)
+  flag('debug.notes.lastPatchError', true)
+
+  // then wait and trigger the collab  debug
+  return await new Promise(resolve => {
+    window.setTimeout(async () => {
+      const collabProvider = get(window, 'cozy.debug.notes.collabProvider')
+      const channel = get(window, 'cozy.debug.notes.channel')
+      const data = { date: new Date().toISOString() }
+
+      data.noteId = get(window, 'cozy.debug.notes.noteId')
+      data.initialVersion = get(window, 'cozy.debug.notes.initialVersion')
+      data.isDirty = get(window, 'cozy.debug.notes.dirtyRef.current')
+      data.returnUrl = get(window, 'cozy.debug.notes.returnUrl')
+
+      const state = collabProvider && collabProvider.getState()
+      const version = collabProvider && state && getVersion(state)
+      const { steps } = (state && sendableSteps(state)) || { steps: [] }
+      data.state = { version, steps, length: steps && steps.length }
+
+      const backoff = channel && channel.backoff
+      const failures = channel && channel.failures
+      const isDirty = channel && channel.isDirty()
+      const isSending = channel && channel.isSending
+      const hasQueue = channel && channel.hasQueuedSteps()
+      const queued = channel && channel.queuedStep
+      const queuedState = queued && queued.state
+      const queuedVersion = queuedState && getVersion(queuedState)
+      const queuedSteps = queued && queued.localSteps && queued.localSteps.steps
+      data.queue = {
+        backoff,
+        failures,
+        isDirty,
+        isSending,
+        hasQueue,
+        version: queuedVersion,
+        steps: queuedSteps,
+        length: queuedSteps && queuedSteps.length
+      }
+
+      let error = get(window, 'cozy.debug.notes.lastPatchError')
+      if (!error && isDirty) {
+        while (channel.isSending) {
+          await new Promise(resolve => window.setTimeout(resolve, 50))
+        }
+        channel.resetBackoff()
+        channel.processQueue()
+        await Promise.race([
+          new Promise(resolve => window.setTimeout(resolve, 2000)),
+          new Promise(resolve => {
+            const fn = async () => {
+              let err = get(window, 'cozy.debug.notes.lastPatchError')
+              while (!err) {
+                await new Promise(resolve => window.setTimeout(resolve, 50))
+                err = get(window, 'cozy.debug.notes.lastPatchError')
+              }
+              error = err
+              resolve()
+            }
+            fn()
+          })
+        ])
+        if (channel.isDirty()) {
+          channel.backoff = backoff
+          channel.failures = failures
+        }
+      }
+      let msg = error && error.message
```

Comment:
Why do we init `msg` as a boolean here?

Reply:
We don't, but I do not want to assert that `error` is a truthy value that does have a `message` property. This is just defensive code in case error is undefined